test/analytic_weights.jl

```diff
@@ -0,0 +1,845 @@
+                                     7.796454326518419e-7 2.7413731130574843e-7 7.79e-7;
+                                     3.441301868580375e-8 1.8149050735676883e-8 3.44e-8;
+                                     1.181434863206281e-9 8.519238822580661e-10 1.18e-9;
+                                     3.115862778487023e-6 5.873759740112369e-7 3.11e-6;
+                                     0.9629196988432743 0.6143391585021523 0.0] rtol = 1e-05
+end
+
+@testset "GLM: Gamma with InverseLink link - AnalyticWeights with $dmethod method" for (dmethod,
+                                                                                        drop) in
+                                                                                       itr
+
+    model = glm(@formula(lot1 ~ 1 + u), clotting, Gamma(), InverseLink(),
+                wts=aweights(clotting.w), method=dmethod, dropcollinear=drop, atol=1e-07,
+                rtol=1e-08)
+    @test deviance(model) ≈ 0.03933389380881642 rtol = 1e-07
+    @test loglikelihood(model) ≈ -43.359078787690514 rtol = 1e-07
+    @test coef(model) ≈ [-0.017217012596343607, 0.015649040406186487] rtol = 1e-07
+    @test stderror(model) ≈ [0.0009144223353860925, 0.0003450913537314497] rtol = 1e-04
+    @test aic(model) ≈ 92.71815757538103 rtol = 1e-07
+    @test bic(model) ≈ 93.30983130738969 rtol = 1e-07
+    @test GLM.momentmatrix(model) ≈ [1900.1063511093867 3058.103199132267;
+                                     -1643.317155973023 -3783.877586404854;
+                                     -420.13783432322964 -1137.7543467296691;
+                                     -981.2887166533023 -2939.6782781526754;
+                                     313.30087123532877 1065.5981029180723;
+                                     -186.60227446859759 -688.353296378139;
+                                     324.34628373045786 1327.9854430687467;
+                                     430.8197010892654 1887.863404915401;
+                                     262.77277766267576 1210.113361381432] rtol = 1e-07
+end
+
+@testset "GLM: Gamma with IdentityLink link - AnalyticWeights with $dmethod method" for (dmethod,
+                                                                                         drop) in
+                                                                                        itr
+
+    model = glm(@formula(lot1 ~ 1 + u), clotting, Gamma(), IdentityLink(),
+                wts=aweights(clotting.w), method=dmethod, dropcollinear=drop,
+                rtol=1e-10, atol=1e-10, minstepfac=0.00001)
+    @test deviance(model) ≈ 1.3435348802929383 rtol = 1e-07
+    @test loglikelihood(model) ≈ -101.19916126647321 rtol = 1e-07
+    @test coef(model) ≈ [86.45700434128152, -15.320695650698417] rtol = 1e-05
+    @test stderror(model) ≈ [16.07962739541372, 3.766841480457265] rtol = 1e-05
+    @test GLM.aic(model) ≈ 208.39832253294642 rtol = 1e-07
+    @test GLM.bic(model) ≈ 208.9899962649551 rtol = 1e-07
```

Comment:
No need for the GLM prefix here and elsewhere.

Reply:
Done